### PR TITLE
✨ 9 - connect login

### PR DESCRIPTION
**Pull request by anncatton**
Sets up auth context
Connects oauth with ego login, for any provider that is configured (facebook is disabled). Tested locally with Google and Orcid
Adds [logout](https://github.com/overture-stack/roadmap/issues/63)
Minor style fixes for navbar, userDropdown

### CODE REVIEW

File: components/pages/user/index.tsx [hunk 1]
```diff
@@ -382,73 +383,78 @@ const StyledPageLayout = styled(PageLayout)`
...
-            display: flex;
-            flex-direction: row;
-            justify-content: space-between;
-            width: 800px;
-            margin-top: 1.5rem;
-            margin-bottom: 0.5rem;
-            padding-bottom: 2.5rem;
-            border-bottom: 1px solid ${theme.colors.grey_3};
-          `}
+          css={() =>
```
Comment (ciaranschutte) [inline components/pages/user/index.tsx:393]:
```
css={css`...`}
```
removing the anon function should be fine

Comment (anncatton) [reply components/pages/user/index.tsx:393]: oh weird why do i have that? thanks for catching!

File: global/hooks/useAuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+export const AuthProvider = ({
+  egoJwt,
+  children,
+}: {
+  egoJwt?: string;
+  children: React.ReactElement;
+}) => {
+  const router = useRouter();
+  const [token, setTokenState] = useState<string>(egoJwt);
```
Comment (ciaranschutte) [inline global/hooks/useAuthContext.tsx:24]: getting a compile error here. Not too sure why yet. There is similar code in ARGO.
a

**Review by ciaranschutte — CHANGES_REQUESTED** [review]
Looks good once that build error is fixed!

**Review by ciaranschutte — APPROVED** [review]
Can't get that type thing working, it's very odd. ARGO has same code and it doesn't complain.
I'd put the type as `any` here and make sure it builds, that way at least we know we should put a type here in the future. It's a small thing to hold back this large feature ticket. @anncatton

**Pull request merged**